src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -129,16 +146,21 @@ public void onClose(int i, String s) {
 
                 waitQuietly(started);
 
-                //We need to get into the project workspace.
-                //The workspace is not known in advance, so we have to execute a cd command.
-                watch.getInput().write(String.format("cd \"%s\"%s", starter.pwd(), NEWLINE).getBytes(StandardCharsets.UTF_8));
-                doExec(watch, launcher.getListener().getLogger(), getCommands(starter));
+                if (starter.pwd() != null) {
+                    // We need to get into the project workspace.
+                    // The workspace is not known in advance, so we have to execute a cd command.
+                    watch.getInput().write(
+                            String.format("cd \"%s\"%s", starter.pwd(), NEWLINE).getBytes(StandardCharsets.UTF_8));
+                }
+                doExec(watch, stream, getCommands(starter));
                 proc = new ContainerExecProc(watch, alive, finished);
                 return proc;
             }
 
             @Override
             public void kill(Map<String, String> modelEnvVars) throws IOException, InterruptedException {
+                // String cookie = modelEnvVars.get(COOKIE_VAR);
+                // TODO we need to use the cookie for something
```

Comment:
need to check the example at https://github.com/jenkinsci/docker-workflow-plugin/blob/71cd8f47b46bbc0b0d0134cb8b3ddcd5c0348733/src/main/java/org/jenkinsci/plugins/docker/workflow/WithContainerStep.java#L249-L278